R/S3_tidy.R

```diff
@@ -1,115 +1,90 @@
+warn_singularities <- function(object) {
+  if (object$rank < object$k) {
+    singularities <- object$k - object$rank
+    what <- ifelse(singularities > 1, " coefficients ", " coefficient ")
+    message(
+      singularities, what,
+      " not defined because the design matrix is rank deficient\n"
+    )
+  }
+}
 
-#' @rdname tidy
-#'
-#' @return A data.frame with with coefficient names, estimates, standard
+#' Tidy an estimatr object
+#' @name estimatr_tidiers
+#' @templateVar class lm_robust
+#' @return A [tibble::tibble()] with columns for coefficient names, estimates, standard
 #' errors, confidence intervals, p-values, degrees of freedom, and the
 #' name of the outcome variable
 #'
-#' @export tidy.lm_robust
+#' @param object An object returned by one of the estimators
+#' @param ... extra arguments (not used)
+#'
 #' @export
+#' @seealso [tidy()], [estimatr::lm_robust()], [estimatr::iv_robust()],  [estimatr::difference_in_means()], [estimatr::horvitz_thompson()]
+#' @family estimatr tidiers
 tidy.lm_robust <- function(object, ...) {
   warn_singularities(object)
   tidy_data_frame(object)
 }
 
-#' @rdname tidy
+#' @rdname estimatr_tidiers
+#' @templateVar class iv_robust
 #'
-#' @export tidy.iv_robust
 #' @export
+#' @family estimatr tidiers
 tidy.iv_robust <- function(object, ...) {
   warn_singularities(object)
   tidy_data_frame(object)
 }
 
-
-#' @rdname tidy
+#' @rdname estimatr_tidiers
+#' @templateVar class difference_in_means
 #'
-#' @export tidy.difference_in_means
 #' @export
-tidy.difference_in_means <- function(object, ...) {
-  tidy_data_frame(object)
-}
-
+#' @family estimatr tidiers
+tidy.difference_in_means <- tidy_data_frame
 
-#' @rdname tidy
+#' @rdname estimatr_tidiers
+#' @templateVar class horvitz_thompson
 #'
-#' @export tidy.horvitz_thompson
 #' @export
-tidy.horvitz_thompson <- function(object, ...) {
-  tidy_data_frame(object)
-}
-
-
-tidy_data_frame <- function(object, digits = NULL) {
-  vec_cols <- c(
-      "coefficients",
-      "std.error",
-      "statistic",
-      "p.value",
-      "conf.low",
-      "conf.high",
-      "df"
-  )
-
-  tidy_mat <- do.call("cbind", lapply(object[vec_cols], as.vector))
-  vec_cols[vec_cols == "coefficients"] <- "estimate"
-  colnames(tidy_mat) <- vec_cols
-  return_frame <- data.frame(
-    term = object[["term"]],
-    tidy_mat,
-    stringsAsFactors = FALSE
-  )
+#' @family estimatr tidiers
+tidy.horvitz_thompson <- tidy_data_frame
 
-  return_frame$outcome <- rep(object[["outcome"]], each = length(object[["term"]]))
-
-  rownames(return_frame) <- NULL
-  return(return_frame)
-}
-
-warn_singularities <- function(object) {
-  if (object$rank < object$k) {
-    singularities <- object$k - object$rank
-    what <- ifelse(singularities > 1, " coefficients ", " coefficient ")
-    message(
-      singularities, what,
-      " not defined because the design matrix is rank deficient\n"
-    )
-  }
+#' @export
+#' @family estimatr tidiers
+tidy.NULL <- function(object, ...) {
```

Comment:
I think you can let generics deal with this?

Reply:
This is all there is in generics for tidy

```
> generics:::tidy
function (x, ...) 
{
    UseMethod("tidy")
}
<bytecode: 0x102698550>
<environment: namespace:generics>
```

Reply:
Right,

```
> tidy(NULL)
Error in UseMethod("tidy") : 
  no applicable method for 'tidy' applied to an object of class "NULL"
```

is a perfectly reasonable behavior, and unless there was/is a good reason for us to change it we should leave it alone.

Reply:
I think it was for an idea when there is a pipe and you are row binding or something? But I agree, I wanted to remove it. Remember why we might need this, @acoppock?

Reply:
No I can't remember any reason we'd need to tidy a null thing.

Reply:
Removed.